examples/animation/animate_decay.py

```diff
@@ -1,17 +1,31 @@
+"""
+=====
+Decay
+=====
+
+This example showcases a decay process with updating x-scale.
+"""
+
 import numpy as np
 import matplotlib.pyplot as plt
 import matplotlib.animation as animation
 
 
 def data_gen(t=0):
-    cnt = 0
-    while cnt < 1000:
-        cnt += 1
+    """Generate data
+
+    Generates a decaying sine wave.
+    """
+    for count in range(1000):
         t += 0.1
         yield t, np.sin(2*np.pi*t) * np.exp(-t/10.)
 
 
 def init():
+    """Create a line
```

Comment:
This is not quite correct. It doesn't create the line, it updates the line's data with the initial value, which is maybe a bit odd, I guess.